portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -37,6 +41,19 @@
 
   @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        Query query = new Query("Task").addSort("timestamp", SortDirection.DESCENDING);
+        int pageLoaded = 0;
+        PreparedQuery results = datastore.prepare(query);
+        for (Entity entity : results.asIterable()) {
+            long id = entity.getKey().getId();
+            String text = (String) entity.getProperty("text");
+            long timestamp = (long) entity.getProperty("timestamp");
+
+            comments.add(text);
```

Comment:
comments should be a function scoped variable. It looks like the intention is to fetch the list of comments from DataStore, then push them into an arrayList. But I don't see this list ever being cleared, so it would appear to append more and more data to the list each time you hit the doGet method.

If you make this variable function scoped, it would go out of scope and get garbage collected after the function returns.

Reply:
Thanks this helped. I assumed I needed a global arrayList because I also returned json in my doPost but I didn't even need to do that because by redirecting the page it calls the doGet. I guess I had struggled to clearly understand the timeline of my code but now it makes sense.

Reply:
😎